src/components/RepoSearchForm.tsx

```diff
@@ -11,33 +11,53 @@ import Search from 'antd/lib/input/Search';
 export const GithubSearchRepositories: React.FC = () => {
     const dispatch: any = useDispatch();
     const [searchName, setSearchName] = useState('');
-    const { list, isLoading } = useSelector((state: StoreState) => state.gitHubRepositories);
-
+    const { list, isLoading, total_repositories_count } = useSelector((state: StoreState) => state.gitHubRepositories);
     const onChangeRepositoryTitle = (e: ChangeEvent<HTMLInputElement>) => setSearchName(e.target.value);
+    const [pageSize, setPageSize] = useState<number>(10);
 
     const onClick = () => {
-        dispatch(searchGithubRepositories(searchName));
+        dispatch(searchGithubRepositories(searchName, 1, pageSize));
+    };
+    const handlePagination = (currentPage: number) => {
+        dispatch(searchGithubRepositories(searchName, currentPage, pageSize));
+    };
+    const handleShowSizeChange = (currentPage: number, pageSize: number) => {
+        setPageSize(pageSize);
+        dispatch(searchGithubRepositories(searchName, currentPage, pageSize));
```

Comment:
Please remove code duplication: 

`onClick`, `handlePagination` and `handleShowSizeChange` functions do the same thing:
```js
dispatch(searchGithubRepositories(searchName, currentPage, pageSize));
```

Move it to a separate function and call with proper parameters.

Reply:
When you move pagination state to the reducer, you'll see that you don't need two functions to initiate search, but only one.